worker/firewaller/firewaller.go

```diff
@@ -333,6 +335,13 @@ func (fw *Firewaller) publishNetworkChanged(change *remoteRelationNetworkChange)
 		logger.Debugf("relation id not found publishing %+v", event)
 		return nil
 	}
+
+	// If the requested ingress is not permitted on the offering side,
+	// mark the relation as in error. It's not an error that requires a
+	// worker restart though.
```

Comment:
say the offerer adjusts their firewall rules, how do you fix this?

Reply:
To be thrashed out, but I'm thinking the relation would be marked as in error and the consumer would need to add-relation again with adjusted ingress.